### PR TITLE
Ensure style-spec is published with mapbox-gl

### PR DESCRIPTION
**Pull request by anandthakker**
Includes a copy of the compiled style-spec bundle under
`dist/style-spec/index.js` in the published mapbox-gl package

### CODE REVIEW

**Comment by anandthakker** [comment]
cc @samanpwbb

File: package.json [hunk 1]
```diff
@@ -141,13 +142,14 @@
     "test-suite": "run-s test-render test-query",
     "test-suite-clean": "find test/integration/{render,query}-tests -mindepth 2 -type d  -not \\( -exec test -e \"{}/style.json\" \\; \\) -print | xargs -t rm -r",
     "test-unit": "build/run-tap --reporter classic --no-coverage test/unit",
+    "test-build": "build/run-tap --no-coverage test/build/**/*.test.js",
```
Comment (jfirebaugh) [inline package.json]: Does this get run in CI?

File: package.json [hunk 2]
```diff
@@ -116,10 +116,11 @@
     "sourceCache": true
   },
   "scripts": {
-    "build-dev": "rollup -c --environment BUILD:dev && build/run-tap --no-coverage test/build/dev.test.js",
+    "build-dev": "rollup -c --environment BUILD:dev",
     "watch-dev": "rollup -c --environment BUILD:dev --watch",
-    "build-min": "rollup -c --environment BUILD:production && build/run-tap --no-coverage test/build/min.test.js",
+    "build-min": "rollup -c --environment BUILD:production",
     "build-css": "postcss -o dist/mapbox-gl.css src/css/mapbox-gl.css",
+    "build-style-spec": "cd src/style-spec && npm run build && cd ../.. && cp -R src/style-spec/dist dist/style-spec",
```
Comment (jfirebaugh) [inline package.json]: Any particular reason for `npm` rather than `yarn`?

Comment (anandthakker) [reply package.json]: Nope, just habit, will change.

File: package.json (hunk 1 above)
Comment (anandthakker) [reply package.json]: Oops, good catch -- accidentally omitted the circle.yml change

File: circle.yml [hunk 3]
```diff
@@ -131,6 +131,8 @@ jobs:
       - run: yarn run build-min
       - run: yarn run build-dev
       - run: yarn run build-css
+      - run: yarn run build-style-spec
+      - run: yarn run test-build
```
Comment (anandthakker) [inline circle.yml]: @jfirebaugh do you see any reason to do `test-build` as a separate workflow step?  I thought it kinda made sense to keep it here, but 🤷‍♂️

Comment (jfirebaugh) [reply circle.yml]: No, adding it here is what I would have done too.

File: package.json (hunk 2 above)
Comment (anandthakker) [reply package.json]: Reverted to `npm` due to https://github.com/mapbox/mbgl-ci-images/issues/17

**Pull request merged**